pkg/controllers/teammembership/teammembership_updater_controller.go

```diff
@@ -121,10 +132,14 @@ func (r *TeamMembershipUpdaterController) Reconcile(ctx context.Context, req ctr
 
 			log.FromContext(ctx).Info("deleting TeamMembership, Team does not have MappedIdpGroup set", "team-membership", teamMembership.Name)
 			err = r.Delete(ctx, teamMembership, &client.DeleteOptions{})
-			return ctrl.Result{}, err
+			if err != nil {
+				return ctrl.Result{}, lifecycle.Failed, err
+			}
+
+			return ctrl.Result{}, lifecycle.Success, nil
 		} else {
 			log.FromContext(ctx).Info("Team does not have MappedIdpGroup set", "team", team.Name)
-			return ctrl.Result{}, nil
+			return ctrl.Result{}, lifecycle.Success, nil
```

Comment:
I think the else block can be removed and you can just log and return success... WDYT?

Reply:
Done